src/Tmds.Ssh/SshConnection.cs

```diff
@@ -0,0 +1,40 @@
+﻿// This file is part of Tmds.Ssh which is released under LGPL-3.0.
+// See file LICENSE for full license details.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+
+namespace Tmds.Ssh
+{
+    public sealed class SshConnection : IDisposable
+    {
+        private readonly SshConnectionSettings _settings;
+        private readonly ILogger _logger;
+
+        public SshConnection(SshConnectionSettings settings, ILogger logger = null)
+        {
+            _settings = settings ?? throw new ArgumentNullException(nameof(settings));
+            _logger = logger;
+        }
+
+        public async Task ConnectAsync(CancellationToken cancellationToken)
+        {
+            // TODO: SshConnectionSettings needs an upper bound time for this method (e.g. SshConnectionSettings.ConnectTimeout)
+        }
+
+        // This method is for doing a clean shutdown which may involve sending some messages over the wire.
+        public async Task DisconnectAsync(CancellationToken cancellationToken)
+        {
+            // TODO: SshConnectionSettings needs an upper bound time for this method (e.g. SshConnectionSettings.DisconnectTimeout)
+
+            // In a finally block, this method calls Dispose.
+        }
+
+        // This method will just cut the connection.
+        public void Dispose()
```

Comment:
Just a thought, if dispose also tries to disconnect cleanly if still connected when called (maybe treat it as is the local connection quit due to a timeout if no other reason is given?), then one could abandon a connection object to the heap/GC and still possibly close cleanly. Kind of a pseudo RAII approach to handling connection objects.

I haven't really thought the effects through completely but it was just something that came to mind.

Reply:
The reason for not trying a clean disconnect is that the user has no cancellationtoken to indicate how long he wants to spend on that. We can have that on an explicit Disconnect method.
```
// All of this may take at most 500ms:
ct = new CancellationTokenSource(500).Token;
using var connection = new SshConnection(...);
await connection.ConnectAsync(ct);
await connection.DoStuffAsync(ct);
await connection.DisconnectAsync(ct); // try a clean disconnect
```
Probably this will change to `IAsyncDisposable`, which suffers from some keyword verbosity:
```
await using var connection = ...
```